File: java/datastructures/lib/src/test/java/codechallenges/repeatedWord/RepeatedWordTest.java
package codechallenges.repeatedWord;

import codechallenges.linkedlist.LinkedListChallenges;
import datastructures.linkedlist.LinkedList;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

import java.util.HashMap;

import static org.junit.jupiter.api.Assertions.assertEquals;

class RepeatedWordTest {

  @Test
  @DisplayName("test one")
  void testOne()
  {
    RepeatedWord sut = new RepeatedWord();
    String string = "Once upon a time, there was a brave princess who...";
    assertEquals(sut.repeatedWord(string), "a");
  }

  @Test
  @DisplayName("test two")
  void testTwo()
  {
    RepeatedWord sut = new RepeatedWord();
    String string = "It was the best of times it was the worst of times, it was the age of wisdom, it was the age of foolishness, it was the epoch of belief, it was the epoch of incredulity, it was the season of Light, it was the season of Darkness, it was the spring of hope, it was the winter of despair, we had everything before us, we had nothing before us, we were all going direct to Heaven, we were all going direct the other way – in short, the period was so far like the present period, that some of its noisiest authorities insisted on its being received, for good or for evil, in the superlative degree of comparison only...";
    assertEquals(sut.repeatedWord(string), "it");
  }

  @Test
  @DisplayName("test three")
  void testThree()
  {
    RepeatedWord sut = new RepeatedWord();
    String string = "It was a queer, sultry summer, the summer they electrocuted the Rosenbergs, and I didn’t know what I was doing in New York...";
    assertEquals(sut.repeatedWord(string), "summer");
  }

  @Test
  @DisplayName("empty string")
  void testFour()
  {
    RepeatedWord sut = new RepeatedWord();
    String string = "";
    assertEquals(sut.repeatedWord(string), "");
  }

}

File: javascript/code-challenges/repeatingWord/__test__/repeatedWord.test.js
const repeatedWord = require('../reapeated-word.js');



describe('Repeated Word', () => {
  it('should return a', () => {
    const str = 'Once upon a time, there was a brave princess who...';
    expect(repeatedWord(str, false, 2)).toBe('a');
  });

  it('should return the', () => {
    let str = 'It was the best of times, it was the worst of times, it was the age of wisdom, it was the age of foolishness, it was the epoch of belief, it was the epoch of incredulity, it was the season of Light, it was the season of Darkness, it was the spring of hope, it was the winter of despair, we had everything before us, we had nothing before us, we were all going direct to Heaven, we were all going direct the other way – in short, the period was so far like the present period, that some of its noisiest authorities insisted on its being received, for good or for evil, in the superlative degree of comparison only...';
    expect(repeatedWord(str, false, 2)).toBe('it');
  });

  it('should return summer', () => {
    const str = 'It was a queer, sultry summer, the summer they electrocuted the Rosenbergs, and I didn’t know what I was doing in New York...';
    expect(repeatedWord(str, false, 2)).toBe('summer');
  });

  it('returns the string object that occurs the most in the array', () => {
    const str = 'Once upon a time, there was a brave princess who...';
    expect(repeatedWord(str, true, 2)).toEqual({key: 'a', value: 2});
  });

  it('returns the string the occurs the most in the array', () => {
    const str = 'o tolerate or endure through the unexpected mishappenings you may encounter from time to time';
    expect(repeatedWord(str, true, 2)).toEqual( {key: 'time', value: 2});
  });

  // it('should return the', () => {
  //   let str = 'It was the best of times, it was the worst of times, it was the age of wisdom, it was the age of foolishness, it was the epoch of belief, it was the epoch of incredulity, it was the season of Light, it was the season of Darkness, it was the spring of hope, it was the winter of despair, we had everything before us, we had nothing before us, we were all going direct to Heaven, we were all going direct the other way – in short, the period was so far like the present period, that some of its noisiest authorities insisted on its being received, for good or for evil, in the superlative degree of comparison only...';
  //   expect(repeatedWord(str, false, 2)).toBe('it');
  // });
});
